### DISCUSSION TITLE
Can I regenerate my Octernship repo

### DISCUSSION THREAD

**Discussion by TiwariAbhishek23** [community/community#57571] [GitHub Education]
### Hi everyone,

During my project work, I made some mistakes. I forgot to include the git ignore file and proceeded to develop the project without it. Now I realize my error, but my commit history is a mess. I would appreciate any guidance or assistance. Thank you.

**Comment by mwt** [comment]
You can go back n commits by running

```sh
git reset HEAD~n --hard
```

But be careful! This is a destructive action. After this, you'll need to force push the changes with `git push --force`.

Make sure the reset command did what you wanted before pushing. So long as you haven't pushed, you can undo the reset by pulling.
